scripts/validateDeploymentUtils.ts

```diff
@@ -0,0 +1,300 @@
+import axios from "axios";
+import hre, { ethers } from "hardhat";
+import * as fs from "fs";
+import * as path from "path";
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { getContractCreationFromEtherscan, getIsGmxDeployer } from "./validateRolesUtils";
+import { execSync } from "child_process";
+import readline from "node:readline";
+
+export interface SignalRoleInfo {
+  account: string;
+  roleKey: string;
+}
+
+export interface ContractInfo {
+  address: string;
+  name: string | null;
+  isCodeValidated: boolean;
+  signalledRoles: string[];
+  unapprovedRoles: string[];
+}
+
+export interface DeploymentInfo {
+  contractName: string;
+  constructorArgs: string[];
+  deploymentTxHash: string;
+}
+
+export async function validateSourceCode(provider: JsonRpcProvider, contractInfo: ContractInfo) {
+  const { contractCreator } = await getContractCreationFromEtherscan(contractInfo.address);
```

Comment:
would be good to cache positive result to avoid sending unnecessary calls. i think they can noticeably slow down updateRoles script